src/v/cluster/partition.cc

```diff
@@ -572,6 +572,7 @@ ss::future<> partition::stop() {
               "Stopping archiver on partition: {}",
               partition_ntp);
             co_await _archiver->stop();
+            _archiver.reset(nullptr);
```

Comment:
does this negate the need for the first commit?

Reply:
in practice, probably yes, but I think it's more correct to not tie the lifetime of the probe to the lifetime of this service.